src/coreclr/jit/gentree.cpp

```diff
@@ -32809,284 +32778,153 @@ GenTree* Compiler::gtFoldExprHWIntrinsic(GenTreeHWIntrinsic* tree)
                     break;
                 }
 
-                default:
+#ifdef TARGET_ARM64
```

Comment:
These are all scenarios we were handling in `VN` but weren't handling here in the more general constant folding path.

We use all of these in the various `CountWhereAllBitsSet`, `IndexOfWhereAllBitsSet`, and related helpers.

Reply:
I guess it'd be nice to declare rules for such constant foldings just once and then reuse from both phases.